### DISCUSSION TITLE
bun build file size is 107.7 KB, esbuild build file size is 7.6 KB

### DISCUSSION THREAD

**Discussion by guest271314** [oven-sh/bun#8015] [General]
I can reduce the `bun build` file size to 50.4 using `--external` to remove two modules not used and `--minify`.

**Comment by Jarred-Sumner** [comment]
Did you try setting `NODE_ENV=production`?

**Reply by guest271314** [reply]
Doesn't change much. I've changed the file minimally. Still 108.9 KB.

**Reply by guest271314** [reply]
Got it down to 68.9 KB with `external: ["mime"]`.

**Reply by guest271314** [reply]
First was able to get it down to 55.4 setting two other imports in `external`, 28 KB with `minify` set to true. 

Got it down further to 6.8 KB by including another import in `external` (`esbuild` does that with `packages: "external"`). I didn't observe any difference with `NODE_ENV=production bun run bun-bundle.js`.

esbuild-bundle.js

```
import { build } from "esbuild";

// Deno-specific workaround for dynamic imports.
const dynamicImport = "./wbn-bundle.js";

try {
  console.log(await build({
    entryPoints: ["src/index.ts"],
    platform: "node",
    outfile: dynamicImport,
    format: "esm",
    packages: "external",
    legalComments: "inline",
    sourcemap: true,
    bundle: true,
    keepNames: true,
    allowOverwrite: true,
  }));
} catch (e) {
  console.log(e);
}
```

bun-bundle.js

```
// import { build } from "esbuild";

// Deno-specific workaround for dynamic imports.
const dynamicImport = "./wbn-bundle.js";

try {
  console.log(
    await Bun.build({
      entrypoints: ["./src/index.ts"],
      outdir: ".",
      sourcemap: "external",
      splitting: false,
      target: "node",
      format: "esm",
     // minify: true,
      external: ["mime", "base32-encode", "wbn-sign-webcrypto", "wbn"],
      naming: {
        entry: "[dir]/wbn-bundle.[ext]",
      },
    }),
  );
} catch (e) {
  console.log(e);
}
```

**Reply by guest271314** [reply]
4.3 KB with `--minify`.

**Reply by guest271314** [reply]
`--target=bun` and `--target=node` appear to yield same result.

**Reply by guest271314** [reply]
Just looking at the file produced.

**Reply by winstxnhdw** [reply]
Yeah same problem here. I am surprised not many people are talking about it.

**Reply by guest271314** [reply]
The last time I checked I got this down to less than what `esbuild` produces 

> 4.3 KB with `--minify`.

**Comment by Jarred-Sumner** [comment]
try setting `NODE_ENV=production` or `--define process.env.NODE_ENV:"'production'"`

**Reply by winstxnhdw** [reply]
Tried this. No externals and I am getting 57kb (Bun) vs 54kb (ESBuild) :/

**Reply by guest271314** [reply]
What happens when you include `--minify`?

**Comment by liudonghua123** [comment]
I have another bun build problem, see https://github.com/oven-sh/bun/issues/10100#issuecomment-2044037563, it seems the js file genreated via bun build not include the dynamic import/require.
